x/lockup/keeper/lock.go

```diff
@@ -146,7 +146,8 @@ func (k Keeper) CreateLockNoSend(ctx sdk.Context, owner sdk.AccAddress, coins sd
 	ID := k.GetLastLockID(ctx) + 1
 	// unlock time is initially set without a value, gets set as unlock start time + duration
 	// when unlocking starts.
-	lock := types.NewPeriodLock(ID, owner, duration, time.Time{}, coins)
+	// the reward receiver is set as the owner by default when creating a lock, and we indicate this by using an empty string.
+	lock := types.NewPeriodLock(ID, owner, "", duration, time.Time{}, coins)
```

Comment:
Should we add an option to have an optional receiver field at lock creation time?

Reply:
Agreed that this might be useful. Due to time constraints, I would like to implement this feature after getting this merged, developing same feature for superfluid staking and then come back if we have capacity. Created an issue here to track this https://github.com/osmosis-labs/osmosis/issues/5338